sdk/webpubsub/azure-messaging-webpubsub/src/test/java/com/azure/messaging/webpubsub/TokenGenerationTest.java

```diff
@@ -46,6 +53,51 @@ public void testTokenGeneration(GetClientAccessTokenOptions tokenOptions, String
         JWTClaimsSet jwtClaimsSet = parse.getJWTClaimsSet();
         assertEquals(expectedSubject, jwtClaimsSet.getSubject());
         assertEquals(expectedRoles, jwtClaimsSet.getClaim("role"));
+        assertEquals(expectedGroups, jwtClaimsSet.getClaim("webpubsub.group"));
+
+    }
+
+    @Test
+    public void testTokenGenerationFromAadCredential() throws ParseException {
+        String endpoint = Configuration.getGlobalConfiguration()
+            .get("WEB_PUB_SUB_ENDPOINT", "http://testendpoint.webpubsubdev.azure.com");
+        WebPubSubServiceClientBuilder webPubSubServiceClientBuilder = new WebPubSubServiceClientBuilder()
+            .endpoint(endpoint)
+            .httpClient(HttpClient.createDefault())
+            .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+            .credential(new DefaultAzureCredentialBuilder().build())
+            .hub("test");
+
+        String connectionString = "Endpoint=http://testendpoint.webpubsubdev.azure.com;"
+                    + "AccessKey=xJItsTUmJB1m+98rVG8YepBvx5BaMnUtGtbGa/oDM+mGyZ=;Version=1.0;Port=8080";
+        if (getTestMode() == TestMode.PLAYBACK) {
+            webPubSubServiceClientBuilder.httpClient(interceptorManager.getPlaybackClient())
+                .connectionString(connectionString);
+        } else if (getTestMode() == TestMode.RECORD) {
+            webPubSubServiceClientBuilder.addPolicy(interceptorManager.getRecordPolicy())
+                .credential(new DefaultAzureCredentialBuilder().build());
+        } else if (getTestMode() == TestMode.LIVE) {
+            webPubSubServiceClientBuilder.credential(new DefaultAzureCredentialBuilder().build());
+        }
+
+        List<String> expectedRoles = Arrays.asList("a", "b");
+        // test for special char case
+        List<String> expectedGroups = Arrays.asList("a&b", "c");
```

Comment:
Also tested when name contains `&`

Reply:
This test needs to use real webpubsub endpoint to generate client token in LIVE/RECORD mode and use recorded json in PLAYBACK mode. So I make `TokenGenerationTest` extends `TestBase`. But this will cause the existing parameterized tokenGeneration tests all needs playback file, otherwise it will have below error. Please let me know if you have any concern or suggestion.
[java.lang.RuntimeException: Missing both new and old playback files](https://dev.azure.com/azure-sdk/public/_build/results?buildId=2093893&view=logs&j=6438ed74-af23-540e-94f3-38d1eefc9e9b&t=2d3af076-e540-5b98-fba0-cc7b8b21203d&l=741)

Reply:
After the changes, does it test the key credential mode, or the AAD credential mode?

You can always skip PLAYBACK, if you only want to have LIVE tests.

Reply:
AAD credential mode. skipping PLAYBACK also needs to extends `TestBase`?